xbmc/interfaces/legacy/InfoTagVideo.cpp

```diff
@@ -193,5 +355,740 @@ namespace XBMCAddon
+    }
+
+    void InfoTagVideo::setIMDBNumber(const String& imdbNumber)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      setIMDBNumberRaw(infoTag, imdbNumber);
+    }
+
+    void InfoTagVideo::setDateAdded(const String& dateAdded)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      setDateAddedRaw(infoTag, dateAdded);
+    }
+
+    void InfoTagVideo::setMediaType(const String& mediaType)
+    {
+      setMediaTypeRaw(infoTag, mediaType);
+    }
+
+    void InfoTagVideo::setShowLinks(std::vector<String> showLinks)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      setShowLinksRaw(infoTag, std::move(showLinks));
+    }
+
+    void InfoTagVideo::setArtists(std::vector<String> artists)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      setArtistsRaw(infoTag, std::move(artists));
+    }
+
+    void InfoTagVideo::setCast(std::vector<const Actor*> actors)
+    {
+      std::vector<SActorInfo> cast;
+      cast.reserve(actors.size());
+      for (const auto& actor : actors)
+        cast.push_back(actor->ToActorInfo());
+
+      {
+        XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+        setCastRaw(infoTag, std::move(cast));
+      }
+    }
+
+    void InfoTagVideo::setResumePoint(double time, double totalTime /* = 0.0 */)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      setResumePointRaw(infoTag, time, totalTime);
+    }
+
+    void InfoTagVideo::addSeason(int number, std::string name /* = "" */)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      addSeasonRaw(infoTag, number, name);
+    }
+
+    void InfoTagVideo::addSeasons(std::vector<Tuple<int, std::string>> namedSeasons)
+    {
+      XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+      addSeasonsRaw(infoTag, namedSeasons);
+    }
+
+    void InfoTagVideo::addVideoStream(const VideoStreamDetail* stream)
+    {
+      if (stream == nullptr)
+        return;
+
+      auto streamDetail = stream->ToStreamDetailVideo();
+      {
+        XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+        addStreamRaw(infoTag, streamDetail);
+      }
+    }
+
+    void InfoTagVideo::addAudioStream(const AudioStreamDetail* stream)
+    {
+      if (stream == nullptr)
+        return;
+
+      auto streamDetail = stream->ToStreamDetailAudio();
+      {
+        XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+        addStreamRaw(infoTag, streamDetail);
+      }
+    }
+
+    void InfoTagVideo::addSubtitleStream(const SubtitleStreamDetail* stream)
+    {
+      if (stream == nullptr)
+        return;
+
+      auto streamDetail = stream->ToStreamDetailSubtitle();
+      {
+        XBMCAddonUtils::GuiLock lock(languageHook, offscreen);
+        addStreamRaw(infoTag, streamDetail);
+      }
+    }
+
+    void InfoTagVideo::setAvailableFanart(const std::vector<const Fanart*>& fanart)
+    {
```

Comment:
I have no idea. I just migrated the existing `ListItem.setAvailableFanart()` to `InfoTagVideo` and made it faster.

Reply:
I'm willing to drop this if there's no need for it.

Reply:
@enen92 what's your verdict?

Reply:
I'd prefer to have it dropped if you agree. I am not sure if this is used at all (sounds like legacy stuff) and if we expose it in some public API it will be harder to remove later

Reply:
Done